thriftbp/circuit_breaker.go

```diff
@@ -0,0 +1,123 @@
+package thriftbp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/sony/gobreaker"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+// FailureRatioBreaker is a circuit breaker based on gobreaker that uses a low-water-mark and
+// % failure threshold to trip.
+type FailureRatioBreaker struct {
+	goBreaker *gobreaker.CircuitBreaker
+
+	name              string
+	minRequestsToTrip int
+	failureThreshold  float64
+	logger            log.Wrapper
+}
+
+// CircuitBreakerConfig represents the configuration for a FailureRatioBreaker
+type CircuitBreakerConfig struct {
+	// Minimum requests that need to be sent during a time period before the breaker is eligible to transition from closed to open.
+	MinRequestsToTrip int
+	// Percentage of requests that need to fail during a time period for the breaker to transition from closed to open.
+	// Represented as a float in [0,1], where .05 means >=5% failures will trip the breaker.
+	FailureThreshold float64
+	// Name for this circuit breaker, mostly used as a prefix to disambiguate logs when multiple cb are used.
+	Name string
+	// EmitStatusMetrics sets whether the failure breaker will regularly update a gauge on the breakers state (closed or open/halfopen)
+	EmitStatusMetrics bool
+	// Logger is the logger to be called when edgecontext parsing failed.
```

Comment:
copy-pasted comment :)

Also, please add a blank line between each comment-field groups (here you didn't group any two fields together so that's between each field).